spec/steps/follow_the_link_spec.rb

```diff
@@ -0,0 +1,65 @@
+require "steps/follow_the_link"
+
+describe Spreewald::Steps::FollowTheLink do
+  it "raises helpful error message if no link is found" do
+    mail_without_links = Mail.new do
+      html_part do
+        body "<html><body>no link</body></html>"
+      end
+
+      text_part do
+        body "no link either"
+      end
+    end
+
+    step = ->() { Spreewald::Steps::FollowTheLink.new(mail_without_links, "first").run }
+    expect(step).to raise_error Spreewald::Steps::FollowTheLink::NoVisitableLinkFound
+  end
+
+  it "finds links in multipart html email" do
+    mail = Mail.new do
+      html_part do
+        body "<html><body><a href='https://www.example.com/abc'>a link</a></body></html>"
+      end
+
+      text_part do
+        body "no link here"
+      end
+    end
+
+    step = Spreewald::Steps::FollowTheLink.new(mail, "first") 
+    expect(step).to receive(:visit_path).with("/abc")
+    step.run
+  end
+
+  it "finds links in html email" do
+    mail = Mail.new do
+      text_part do
+        body "my link: https://www.example.com/abc"
```

Comment:
Maybe a test with more than one link and "second" would be a good idea, to check if it finds the right one and not only the first one.

Reply:
Added a spec for it :)